server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -270,6 +303,77 @@ public ClusterMetadataManifest writeIncrementalMetadata(
         return manifest;
     }
 
+    /**
+     * Uploads provided ClusterState's global Metadata to remote store in parallel.
+     * The call is blocking so the method waits for upload to finish and then return.
+     *
+     * @param clusterState current ClusterState
+     * @return String file name where globalMetadata file is stored.
+     */
+    private String writeGlobalMetadata(ClusterState clusterState)
```

Comment:
Metadata has ```CoordinationMetadata``` which contains information about voting configuration, term etc. Have you tested this code with download & restore flow on single node cluster. I think it wouldn't work when you will seed the single node domain with nodeId2 (after QL recovery) while this voting configuration would have nodeId1. Either don't upload it alogether or you need to skip it during restore.

Reply:
Also fields like cluster_uuid_committed

Reply:
I guess restore PR isn't restoring it, it's simpler that way to serialise and upload while handling restores appropriately